### PR TITLE
Implement parallel shortcuts

### PR DESCRIPTION
**Pull request by antoinewdg**
As suggested in #156, this PR introduces a trait that enables a shorter version of `into_par_iter().map()`.

I did not know where to put it so I put it in `types`, I don't think it belongs here.

### CODE REVIEW

**Comment by Luthaf** [comment]
I think we can put this in the `utils` module. How about calling the trait `ParrallelMap`?

**Comment by antoinewdg** [comment]
> How about calling the trait ParrallelMap?

With the current name we could add other shortcuts as `par_for_each` if needed.

The `utils` module is private, is that an issue ?

**Comment by Luthaf** [comment]
> The utils module is private, is that an issue ?

I see this trait as "for internal use" mainly, so this is fine. If it is useful to others, we can try to add it to rayon directly.

**Comment by antoinewdg** [comment]
I moved the trait to `utils`. What's your final take on the name ?
About moving it to rayon, I don't know if this is something every user of rayon would want.

File: src/core/src/utils/parallel_shortcuts.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+use std::ops::Range;
+
+use rayon::prelude::*;
+use rayon::iter::{Map, MapFn};
+
+
+/// Utility trait that adds shortcuts for
```
Comment (Luthaf) [inline src/core/src/utils/parallel_shortcuts.rs]: Can you put the doc on one line?

File: src/core/src/utils/parallel_shortcuts.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+pub trait ParallelShortcuts : IntoParallelIterator + Sized {
+
+    /// Shortcut for `into_par_iter().map()`
+    fn par_map<F, R>(self, map_op: F) -> Map<Self::Iter, MapFn<F>>
+        where F: Fn(Self::Item) -> R + Sync, R: Send{
+        self.into_par_iter().map(map_op)
+    }
+}
+
+impl<T> ParallelShortcuts for Range<T>
```
Comment (Luthaf) [inline src/core/src/utils/parallel_shortcuts.rs]: Can we implement ParallelShortcuts for T: IntoParallelIterator?

**Comment by antoinewdg** [comment]
Done :)

**Pull request merged**